.github/workflows/release-workflow-dispatcher.yml

```diff
@@ -0,0 +1,321 @@
+          nix-shell --pure --run "echo" ./shell.nix
+
+      - name: Get dependencies version from develop
+        id: extract-versions
+        run: |
+          GITHUB_API_URL="https://raw.githubusercontent.com"
+
+          # Define output file for GitHub Actions
+          OUTPUT_FILE="$GITHUB_OUTPUT"
+
+          # Fetch and extract versions
+          echo "Fetching versions from respective repositories..."
+
+          echo "localpv_provisioner_version=$(curl -s $GITHUB_API_URL/openebs/dynamic-localpv-provisioner/refs/heads/develop/deploy/helm/charts/Chart.yaml | yq '(.version | sub("-develop$"; ""))')" >> "$OUTPUT_FILE"
+          echo "zfs_localpv_version=$(curl -s $GITHUB_API_URL/openebs/zfs-localpv/refs/heads/develop/deploy/helm/charts/Chart.yaml | yq '(.version | sub("-develop$"; ""))')" >> "$OUTPUT_FILE"
+          echo "lvm_localpv_version=$(curl -s $GITHUB_API_URL/openebs/lvm-localpv/refs/heads/develop/deploy/helm/charts/Chart.yaml | yq '(.version | sub("-develop$"; ""))')" >> "$OUTPUT_FILE"
+          echo "mayastor_version=$(curl -s $GITHUB_API_URL/openebs/mayastor-extensions/refs/heads/develop/chart/Chart.yaml | yq '(.version | sub("-develop$"; ""))')" >> "$OUTPUT_FILE"
+
+      - name: Debug extracted versions
+        run: |
+          echo "Extracted LocalPV Provisioner Version: ${{ steps.extract-versions.outputs.localpv_provisioner_version }}"
+          echo "Extracted ZFS LocalPV Version: ${{ steps.extract-versions.outputs.zfs_localpv_version }}"
+          echo "Extracted LVM LocalPV Version: ${{ steps.extract-versions.outputs.lvm_localpv_version }}"
+          echo "Extracted Mayastor Version: ${{ steps.extract-versions.outputs.mayastor_version }}"
+
+      - name: Test the chart version updater script
+        run: |
+          nix-shell --pure --run "./scripts/helm/test-update-chart-version.sh" ./shell.nix
+
+      - name: Modify the chart version based on the branch name for develop
+        run: |
+          branch_name=${{ github.ref_name }}
+          nix-shell --pure --run "./scripts/helm/update-chart-version.sh --branch $branch_name --type develop --localpv-provisioner-version ${LOCALPV_PROVISIONER_VERSION} --zfs-localpv-version ${ZFS_LOCALPV_VERSION} --lvm-localpv-version ${LVM_LOCALPV_VERSION} --mayastor-version ${MAYASTOR_VERSION}" ./shell.nix
+        env:
+          LOCALPV_PROVISIONER_VERSION: ${{ steps.extract-versions.outputs.localpv_provisioner_version }}
+          ZFS_LOCALPV_VERSION: ${{ steps.extract-versions.outputs.zfs_localpv_version }}
+          LVM_LOCALPV_VERSION: ${{ steps.extract-versions.outputs.lvm_localpv_version }}
+          MAYASTOR_VERSION: ${{ steps.extract-versions.outputs.mayastor_version }}
+
+      - name: Pre-populate helm nix-shell
+        run: nix-shell ./scripts/helm/shell.nix --run "echo"
+
+      - name: Generate image list
+        run: nix-shell ./scripts/helm/shell.nix --run "./scripts/helm/images.sh generate --dependency-update --exit-code"
```

Comment:
Don't think we want to use exit-code here since we actually want to modify the metadata here, otherwise this would fail?